codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -446,16 +446,18 @@ private static AsciiString toHttp2Path(URI uri) {
         return path.isEmpty() ? EMPTY_REQUEST_PATH : new AsciiString(path);
     }
 
-    private static void setHttp2Authority(String authority, Http2Headers out) {
+    static void setHttp2Authority(String authority, Http2Headers out) {
         // The authority MUST NOT include the deprecated "userinfo" subcomponent
         if (authority != null) {
             int endOfUserInfo = authority.indexOf('@');
             if (endOfUserInfo < 0) {
                 out.authority(new AsciiString(authority));
-            } else if (endOfUserInfo + 1 < authority.length()) {
-                out.authority(new AsciiString(authority.substring(endOfUserInfo + 1)));
             } else {
-                throw new IllegalArgumentException("authority: " + authority);
+                int start = endOfUserInfo + 1;
+                if (start >= authority.length()) {
```

Comment:
the overflow condition should also be checked by the `AsciiString` constructor ... I guess this is just to provide a more descriptive exception message?